src/MSBuild/XMake.cs

```diff
@@ -266,6 +274,57 @@ string[] args
             return exitCode;
         }
 
+
+        /// <summary>
+        /// Returns true if arguments allows or make sense to leverage msbuild server.
+        /// </summary>
+        /// <remarks>
+        /// Will not throw. If arguments processing fails, we will not run it on server - no reason as it will not run any build anyway.
+        /// </remarks>
+        private static bool CanRunServerBasedOnCommandLineSwitches(
+#if FEATURE_GET_COMMANDLINE
+            string commandLine
+#else
+            string[] commandLine
+#endif
+            )
+        {
+            bool canRunServer = true;
+            try
+            {
+                GatherAllSwitches(commandLine, out var switchesFromAutoResponseFile, out var switchesNotFromAutoResponseFile, out string fullCommandLine);
```

Comment:
I think this shouldn't have much perf impact because we only would do this once at the start of the build, but I'm wondering about small builds where even a little extra perf can matter. Can we reuse what we parsed here?

Reply:
I measured it and it is under 1ms. 
But that being said, whole argument parsing would benefit refactoring - for clarity. I would strongly suggest to split argument parsing from interpreting for example. 
But that shall be addressed in different PR.